src/main/java/org/session9/homework/fitnesstrackingsystem/User.java

```diff
@@ -0,0 +1,92 @@
+package org.session9.homework.fitnesstrackingsystem;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+public class User {
+
+    private String name;
+    private int age;
+    private List<Workout> workouts;
+    private Map<String, Double> weeklyGoals;
+
+    public User(String name, int age) {
+        this.name = name;
+        this.age = age;
+        workouts = new ArrayList<>();
+        weeklyGoals = new HashMap<>();
+    }
+
+    public String getName() {
+        return name;
+    }
+
+    public int getAge() {
+        return age;
+    }
+
+    public void addWorkout(Workout workout) {
+        workouts.add(workout);
+    }
+
+    public List<Workout> getWorkouts() {
+        return workouts;
+    }
+
+    public void setWeeklyGoal(String workoutType, Double goal) {
+        weeklyGoals.putIfAbsent(workoutType, goal);
+    }
+
+    public double getTotalDistance() {
+        double totalDistance = 0;
+        for (Workout workout : workouts) {
+            totalDistance += workout.getDistance();
+        }
+        return totalDistance;
+    }
+
+    public double getTotalDuration() {
+        double totalDuration = 0;
+        for (Workout workout : workouts) {
+            totalDuration += workout.getDuration();
+        }
+        return totalDuration;
+    }
+
+    public double getTotalLiftedWeigth() {
+        double totalWeightLifted = 0;
+        for (Workout workout : workouts) {
+            totalWeightLifted += workout.getLiftedWeight();
+        }
+        return totalWeightLifted;
+    }
+
+    public void generateWeeklyReport() {
+        System.out.println("Weekly report for " + name + ", age : " + age);
+        System.out.println("Total workouts : " + workouts);
+        System.out.println("\nTotal distance : " + getTotalDistance() + " km");
+        System.out.println("Total duration : " + getTotalDuration() + " min");
+        System.out.println("Total lifted weight :" + getTotalLiftedWeigth() + " kg");
+    }
+
+    public void checkGoals() {
+        for (String workoutType : weeklyGoals.keySet()) {
+            double goal = weeklyGoals.get(workoutType);
+            String achievement = null;
+            if (workoutType.equals("running")) {
```

Comment:
We can use a switch maybe in this case? Having a switch will make our code easier to scale in time. What if our client wants to add 3 workout types? 
Also, `running`, `yoga`, and `weightLifting` can be declared as constants and we can maybe make `equalsIgnoreCase` compare for this particular situation. We should not care if it's `yoga` or `Yoga`, the result of the comparison should be true in both scenarios.